timemachine/cpp/src/types.hpp

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+// CoordsType defines both the coordinates and the boxes
+typedef double CoordsType;
+typedef double ParamsType;
```

Comment:
One potential thing we could do here if we only really care about using double for testing purposes (@proteneer perhaps I misunderstood the purpose beyond developer utility for this being double) we could have a compile time flag to switch this to float and get the 2% up.